zerver/lib/bulk_create.py

```diff
@@ -44,11 +53,15 @@ def bulk_create_users(
         )
         profiles_to_create.append(profile)
 
-    if realm.email_address_visibility == Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+    if (
+        realm_user_default.email_address_visibility
+        == RealmUserDefault.EMAIL_ADDRESS_VISIBILITY_EVERYONE
+    ):
         UserProfile.objects.bulk_create(profiles_to_create)
     else:
         for user_profile in profiles_to_create:
             user_profile.email = user_profile.delivery_email
+            user_profile.email_address_visibility = realm_user_default.email_address_visibility
```

Comment:
It's kinda weird that we're both passing this value into `create_user_profile` and then applying it here.  I think the structural bug here is that `create_user_profile` ignores `RealmUserDefault` in general, and this bulk-creation code path ends up ignoring it as a result.  

I think the correct fix here would be to just do a prep commit to have the `bulk_create_users` code path do this:
```
    if bot_type is None:
        realm_user_default = RealmUserDefault.objects.get(realm=realm)
        copy_default_settings(realm_user_default, user_profile)
```

in the loop generating `profiles_to_create`, instead of adding this line.